tests/utils/browser.rs

```diff
@@ -0,0 +1,177 @@
+use image::GenericImageView;
+use playwright::{api::Page, Playwright};
+use std::path::{Path, PathBuf};
+
+pub struct BrowserHelpers;
+
+impl BrowserHelpers {
+    /// Initialize playwright and return browser page
+    pub async fn setup_browser() -> Result<(Playwright, Page), Box<dyn std::error::Error>> {
+        use std::path::Path;
+
+        let playwright = Playwright::initialize().await?;
+        // Skip playwright.prepare() - don't install browsers, use system ones
+
+        // Check for BROWSER_VISIBLE environment variable to run in non-headless mode
+        let headless = std::env::var("BROWSER_VISIBLE").is_err();
+
+        if !headless {
+            println!("🌐 Running browser in VISIBLE mode for debugging");
+        }
+
+        // Use system Chrome - confirmed working on Mac M1 ARM64
+        let chromium = playwright.chromium();
+        let browser = chromium
+            .launcher()
+            .headless(headless)
+            .executable(Path::new(
+                "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
+            ))
+            .launch()
+            .await?;
+        let context = browser.context_builder().build().await?;
+        let page = context.new_page().await?;
+
+        Ok((playwright, page))
+    }
+
+    /// Take a screenshot and compare it to a stored reference image
+    /// Optionally grey out elements that should be ignored in comparison
+    /// Automatically prints comparison results with standardized messages
+    ///
+    /// # Arguments
+    /// * `page` - The Playwright page to screenshot
+    /// * `test_name` - Name for the test (used for file naming and messages)
+    /// * `selectors_to_grey` - CSS selectors for elements to grey out before comparison
+    ///
+    /// # Returns
+    /// * `Ok(())` - Screenshot comparison completed (prints result automatically)
+    /// * `Err(...)` - Error taking screenshot or processing images
+    ///
+    /// # Example
+    /// ```ignore
+    /// BrowserHelpers::screenshot_compare(
+    ///     &page,
+    ///     "login_page",
+    ///     &[".timestamp", ".session-id"]
+    /// ).await?;
+    /// ```
+    pub async fn screenshot_compare(
```

Comment:
Treat screenshots not matching as an Err --- only if everything matches is the result Ok.